frontend/packages/integration-tests-cypress/tests/crud/k8-openshift-cruds.spec.ts

```diff
@@ -87,16 +87,17 @@ describe('Kubernetes resource CRUD operations', () => {
         cy.testA11y(`YAML Editor for ${kind}: ${name}`);
         let newContent;
         // get, update, and set yaml editor content.
-        return yamlEditor.getEditorContent().then((content) => {
+        yamlEditor.getEditorContent().then((content) => {
           newContent = _.defaultsDeep(
             {},
             { metadata: { name, labels: { [testLabel]: testName } } },
             safeLoad(content),
           );
-          yamlEditor.setEditorContent(safeDump(newContent, { sortKeys: true }));
           cy.log('creates a new resource instance');
-          yamlEditor.clickSaveCreateButton();
-          cy.get(errorMessage).should('not.exist');
+          yamlEditor.setEditorContent(safeDump(newContent, { sortKeys: true })).then(() => {
```

Comment:
Is the `then` necessary here? Does Cypress not block automatically?

Reply:
Cypress is asynchronous and they show examples of nested `.then()`'s

Reply:
I'm still not sure it's needed. Are you looking at https://docs.cypress.io/api/commands/window.html#Check-a-custom-property ?

That's reading a variable set inside the `then` later in the test, which doesn't apply here.

Reply:
We've seen flakes around saving from yamlEditor, this way ensures the new content is set...then, `yamlEditor.clickSaveCreateButton();`.